### ISSUE TITLE
Better support for range requests

### ISSUE THREAD

**Issue by yagee-de** [UW-Madison-Library/ocfl-java#1]
For large video or PDF files I need to response to [range requests](https://tools.ietf.org/html/rfc7233#section-3).
[OcflObjectReader](https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-api/src/main/java/edu/wisc/library/ocfl/api/OcflObjectReader.java) should allow access to physical [Path](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html) on the underlying filesystem to allow efficient handling of those requests.

**Comment by pwinckles** [comment]
How about I modify the [FileDetails](https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-api/src/main/java/edu/wisc/library/ocfl/api/model/FileDetails.java) object that's returned as part of a `describeObject`/`describeVersion` request to include the path to the file relative to the storage root?

For example, let's you have object `obj123` and v1 of the object contains the file `dir1/file1.mov`. The `FileDetails` object in this case would look something like the following:

```json
{
  "filePath": "dir1/file1.mov",
  "storagePath": "obj123/v1/content/dir1/file1.mov",
  "fixity": {
    "sha512": "..."
  }
}
```

Would that work for you?

**Comment by yagee-de** [comment]
Hi this sounds like a way to go, but I need to get the storage path of `obj123` via [ObjectDetails](https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-api/src/main/java/edu/wisc/library/ocfl/api/model/ObjectDetails.java) and/or via [OcflRepository](https://github.com/UW-Madison-Library/ocfl-java/blob/master/ocfl-java-api/src/main/java/edu/wisc/library/ocfl/api/OcflRepository.java). In the end I need an absolute path to serve the content, so handling with the triple of `OcflRepository`, `ObjectDetails` and `FileDetails` could make the code ugly. Maybe all three could implement an Interface e.g. `PathResolveable` with two methods `Path getRelativePath()` and `Path getAbsolutePath()`?

**Comment by pwinckles** [comment]
I cannot add absolute paths to the API because this does not make sense for all implementations, such as the Amazon S3 implementation that I recently started working on.

Just so I'm sure we're on the same page, there are three paths of interest.

1. **Storage root path:** The path to the OCFL storage root directory that contains all of the objects.
1. **Object root path:** The path from the storage root to the root directory of an object. In the case of the `ocfl-java` library, this path is dictated by the `ObjectIdPathMapper` that is used.
1. **File path:** The path from the object root to the file's location within a version's content directory.

In the `FileDetails` solution that I purposed, `storagePath` is the equivalent of `objectRootPath`/`filePath`. That is to say, it is the path to a file within an object relative to the storage root. (As an aside, this value is also the key that the file is stored under in S3.) All your code would need to do is resolve the `storagePath` with the path to the storage root.
